src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -324,5 +324,18 @@ public static bool TryGetEscapedGuid(ReadOnlySpan<byte> source, out Guid value)
             value = default;
             return false;
         }
+
+        public static char GetFloatingPointStandardParseFormat(ReadOnlySpan<byte> span)
+        {
+            foreach (byte token in span)
```

Comment:
Would an indexer be faster (no enumerator would need to be created). Also perhaps faster to go from end to start assuming "E" is closer the end.

Reply:
Yup, it's probably faster - we use indexers in other areas (e.g. writing arrays and lists). +1 to starting from the end.